constellations/gemini-admin/src/user/userInput.js

```diff
@@ -0,0 +1,92 @@
+"use strict";
+require("./userInput.sass");
+import React from "react";
+import { InputText, emptyValidator } from "../utils/inputText.js";
+
+export class UserInput extends React.Component {
+    render = () => {
+        const handleChange = this.props.handleChange;
+
+        return (
+            <div id="view-user-inputs">
+                <InputText
+                    label="First Name"
+                    description="Enter your first name"
+                    required={true}
+                    value={this.props.firstName}
+                    onChangeCallback={(e) => handleChange(e, "firstName")}
+                    validators={[emptyValidator("name")]}
+                />
+
+                <InputText
+                    label="Middle Name"
+                    description="Enter your middle name if applicable"
+                    value={this.props.middleName}
+                    onChangeCallback={(e) => handleChange(e, "middleName")}
+                />
+
+                <InputText
+                    label="Last Name"
+                    description="Enter your last name"
+                    required={true}
+                    value={this.props.lastName}
+                    onChangeCallback={(e) => handleChange(e, "lastName")}
+                    validators={[emptyValidator("name")]}
+                />
+
+                <InputText
+                    label="Email"
+                    description="Enter your email address"
+                    required={true}
+                    value={this.props.email}
+                    onChangeCallback={(e) => handleChange(e, "email")}
+                    validators={[
+                        emptyValidator("email"),
+                        {
+                            validate: (email) =>
+                                /^[^( @)]+@[^( @)]+\.[^( @)]+$/.test(email),
```

Comment:
it's kind of wonky having pattern matching regex inside some HTML block.
Can we use a `validateEmail` function as a utility function?

So this block looks like this:
```
validate: (email) => validateEmail(email)
```